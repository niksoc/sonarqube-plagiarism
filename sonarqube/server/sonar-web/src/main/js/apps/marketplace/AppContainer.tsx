/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { connect } from 'react-redux';
import AdminContext from '../../app/components/AdminContext';
import { getAppState, getGlobalSettingValue, Store } from '../../store/rootReducer';
import { EditionKey } from '../../types/editions';
import App from './App';

interface OwnProps {
  location: { pathname: string; query: T.RawQuery };
}

interface StateToProps {
  currentEdition?: EditionKey;
  standaloneMode?: boolean;
  updateCenterActive: boolean;
}

const mapStateToProps = (state: Store) => {
  const updateCenterActive = getGlobalSettingValue(state, 'sonar.updatecenter.activate');
  return {
    currentEdition: getAppState(state).edition as EditionKey, // TODO: Fix once AppState is no longer ambiant.
    standaloneMode: getAppState(state).standalone,
    updateCenterActive: Boolean(updateCenterActive && updateCenterActive.value === 'true')
  };
};

const WithAdminContext = (props: StateToProps & OwnProps) => (
  <AdminContext.Consumer>
    {({ fetchPendingPlugins, pendingPlugins }) => (
      <App fetchPendingPlugins={fetchPendingPlugins} pendingPlugins={pendingPlugins} {...props} />
    )}
  </AdminContext.Consumer>
);

export default connect(mapStateToProps)(WithAdminContext);
