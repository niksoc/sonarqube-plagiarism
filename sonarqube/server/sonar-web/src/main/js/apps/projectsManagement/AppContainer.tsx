/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { connect } from 'react-redux';
import { changeProjectDefaultVisibility } from '../../api/permissions';
import { receiveOrganizations } from '../../store/organizations';
import { fetchOrganization } from '../../store/rootActions';
import { getAppState, getCurrentUser, getOrganizationByKey, Store } from '../../store/rootReducer';
import App from './App';

interface StateProps {
  appState: { defaultOrganization: string; qualifiers: string[] };
  currentUser: T.LoggedInUser;
  organization?: T.Organization;
}

interface DispatchProps {
  fetchOrganization: (organization: string) => void;
  onVisibilityChange: (organization: T.Organization, visibility: T.Visibility) => void;
}

interface OwnProps {
  onRequestFail: (error: any) => void;
  organization: T.Organization;
}

class AppContainer extends React.PureComponent<OwnProps & StateProps & DispatchProps> {
  componentDidMount() {
    // if there is no organization, that means we are in the global scope
    // let's fetch defails for the default organization in this case
    if (!this.props.organization || !this.props.organization.projectVisibility) {
      this.props.fetchOrganization(this.props.appState.defaultOrganization);
    }
  }

  handleOrganizationUpgrade = () => {
    this.props.fetchOrganization(this.props.organization.key);
  };

  handleVisibilityChange = (visibility: T.Visibility) => {
    if (this.props.organization) {
      this.props.onVisibilityChange(this.props.organization, visibility);
    }
  };

  render() {
    const { organization } = this.props;

    if (!organization) {
      return null;
    }

    const topLevelQualifiers = organization.isDefault ? this.props.appState.qualifiers : ['TRK'];
    const { actions = {} } = organization;

    return (
      <App
        currentUser={this.props.currentUser}
        hasProvisionPermission={actions.provision}
        onOrganizationUpgrade={this.handleOrganizationUpgrade}
        onVisibilityChange={this.handleVisibilityChange}
        organization={organization}
        topLevelQualifiers={topLevelQualifiers}
      />
    );
  }
}

const mapStateToProps = (state: Store, ownProps: OwnProps) => ({
  appState: getAppState(state),
  currentUser: getCurrentUser(state) as T.LoggedInUser,
  organization:
    ownProps.organization || getOrganizationByKey(state, getAppState(state).defaultOrganization)
});

const onVisibilityChange = (organization: T.Organization, visibility: T.Visibility) => (
  dispatch: Function
) => {
  const currentVisibility = organization.projectVisibility;
  dispatch(receiveOrganizations([{ ...organization, projectVisibility: visibility }]));
  changeProjectDefaultVisibility(organization.key, visibility).catch(() => {
    dispatch(receiveOrganizations([{ ...organization, projectVisibility: currentVisibility }]));
  });
};

const mapDispatchToProps = (dispatch: Function) => ({
  fetchOrganization: (key: string) => dispatch(fetchOrganization(key)),
  onVisibilityChange: (organization: T.Organization, visibility: T.Visibility) =>
    dispatch(onVisibilityChange(organization, visibility))
});

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
