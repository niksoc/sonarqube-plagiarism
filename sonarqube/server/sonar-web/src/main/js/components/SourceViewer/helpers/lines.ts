/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { intersection } from 'lodash';

export function optimizeHighlightedSymbols(
  symbolsForLine: string[] = [],
  highlightedSymbols: string[] = []
): string[] | undefined {
  const symbols = intersection(symbolsForLine, highlightedSymbols);

  return symbols.length ? symbols : undefined;
}

export function optimizeLocationMessage(
  highlightedLocationMessage: { index: number; text: string | undefined } | undefined,
  optimizedSecondaryIssueLocations: T.LinearIssueLocation[]
) {
  return highlightedLocationMessage != null &&
    optimizedSecondaryIssueLocations.some(
      location => location.index === highlightedLocationMessage.index
    )
    ? highlightedLocationMessage
    : undefined;
}

export function optimizeSelectedIssue(selectedIssue: string | undefined, issuesForLine: T.Issue[]) {
  return selectedIssue !== undefined && issuesForLine.find(issue => issue.key === selectedIssue)
    ? selectedIssue
    : undefined;
}
