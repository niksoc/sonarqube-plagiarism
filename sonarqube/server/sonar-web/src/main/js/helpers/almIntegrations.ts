/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isLoggedIn } from './users';

export function getAlmMembersUrl(key: string, url: string): string {
  if (!url.endsWith('/')) {
    url += '/';
  }
  if (isGithub(key)) {
    return url.replace('github.com/', 'github.com/orgs/') + 'people';
  }
  return url + 'profile/members';
}

export function getUserAlmKey(user: T.CurrentUser) {
  return isLoggedIn(user) && user.externalProvider
    ? sanitizeAlmId(user.externalProvider)
    : undefined;
}

export function hasAdvancedALMIntegration(user: T.CurrentUser) {
  return (
    isLoggedIn(user) && (isBitbucket(user.externalProvider) || isGithub(user.externalProvider))
  );
}

export function isBitbucket(almKey?: string): boolean {
  return almKey ? almKey.startsWith('bitbucket') : false;
}

export function isGithub(almKey?: string): boolean {
  return almKey === 'github';
}

export function isVSTS(almKey?: string): boolean {
  return almKey === 'microsoft';
}

export function sanitizeAlmId(almKey: string) {
  if (isBitbucket(almKey)) {
    return 'bitbucket';
  }
  return almKey;
}
