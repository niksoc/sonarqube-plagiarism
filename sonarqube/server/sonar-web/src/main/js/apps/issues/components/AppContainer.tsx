/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { uniq } from 'lodash';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { lazyLoad } from 'sonar-ui-common/components/lazyLoad';
import { searchIssues } from '../../../api/issues';
import { getOrganizations } from '../../../api/organizations';
import throwGlobalError from '../../../app/utils/throwGlobalError';
import { parseIssueFromResponse } from '../../../helpers/issues';
import { receiveOrganizations } from '../../../store/organizations';
import {
  areThereCustomOrganizations,
  getCurrentUser,
  getMyOrganizations,
  Store
} from '../../../store/rootReducer';

interface StateProps {
  currentUser: T.CurrentUser;
  userOrganizations: T.Organization[];
}

const mapStateToProps = (state: Store): StateProps => ({
  currentUser: getCurrentUser(state),
  userOrganizations: getMyOrganizations(state)
});

const fetchIssueOrganizations = (organizationKeys: string[]) => (dispatch: Dispatch) => {
  if (!organizationKeys.length) {
    return Promise.resolve();
  }

  return getOrganizations({ organizations: organizationKeys.join() }).then(
    response => dispatch(receiveOrganizations(response.organizations)),
    throwGlobalError
  );
};

const fetchIssues = (query: T.RawQuery, requestOrganizations = true) => (
  // use `Function` to be able to do `dispatch(...).then(...)`
  dispatch: Function,
  getState: () => Store
) => {
  const organizationsEnabled = areThereCustomOrganizations(getState());
  return searchIssues({ ...query, additionalFields: '_all' })
    .then(response => {
      const parsedIssues = response.issues.map(issue =>
        parseIssueFromResponse(issue, response.components, response.users, response.rules)
      );
      return { ...response, issues: parsedIssues };
    })
    .then(response => {
      const organizationKeys = uniq([
        ...response.issues.map(issue => issue.organization),
        ...(response.components || []).map(component => component.organization)
      ]);
      return organizationsEnabled && requestOrganizations
        ? dispatch(fetchIssueOrganizations(organizationKeys)).then(() => response)
        : response;
    })
    .catch(throwGlobalError);
};

interface DispatchProps {
  fetchIssues: (query: T.RawQuery, requestOrganizations?: boolean) => Promise<void>;
}

// have to type cast this, because of async action
const mapDispatchToProps = { fetchIssues: fetchIssues as any } as DispatchProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lazyLoad(() => import('./App'), 'IssuesAppContainer'));
