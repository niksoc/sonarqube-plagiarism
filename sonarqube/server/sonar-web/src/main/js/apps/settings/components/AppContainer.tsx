/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { find } from 'lodash';
import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import { connect } from 'react-redux';
import { WithRouterProps } from 'react-router';
import { translate } from 'sonar-ui-common/helpers/l10n';
import Suggestions from '../../../app/components/embed-docs-modal/Suggestions';
import { getSettingsAppDefaultCategory, Store } from '../../../store/rootReducer';
import '../side-tabs.css';
import { fetchSettings } from '../store/actions';
import '../styles.css';
import { ADDITIONAL_CATEGORIES } from './AdditionalCategories';
import AllCategoriesList from './AllCategoriesList';
import CategoryDefinitionsList from './CategoryDefinitionsList';
import { CATEGORY_OVERRIDES } from './CategoryOverrides';
import PageHeader from './PageHeader';

interface Props {
  component?: T.Component;
  defaultCategory: string;
  fetchSettings(component?: string): Promise<void>;
}

interface State {
  loading: boolean;
}

export class App extends React.PureComponent<Props & WithRouterProps, State> {
  mounted = false;
  state: State = { loading: true };

  componentDidMount() {
    this.mounted = true;
    this.fetchSettings();
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.component !== this.props.component) {
      this.fetchSettings();
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  fetchSettings = () => {
    const { component } = this.props;
    this.props.fetchSettings(component && component.key).then(this.stopLoading, this.stopLoading);
  };

  stopLoading = () => {
    if (this.mounted) {
      this.setState({ loading: false });
    }
  };

  render() {
    if (this.state.loading) {
      return null;
    }

    const { query } = this.props.location;
    const originalCategory = (query.category as string) || this.props.defaultCategory;
    const overriddenCategory = CATEGORY_OVERRIDES[originalCategory.toLowerCase()];
    const selectedCategory = overriddenCategory || originalCategory;
    const foundAdditionalCategory = find(ADDITIONAL_CATEGORIES, c => c.key === selectedCategory);
    const isProjectSettings = this.props.component;
    const shouldRenderAdditionalCategory =
      foundAdditionalCategory &&
      ((isProjectSettings && foundAdditionalCategory.availableForProject) ||
        (!isProjectSettings && foundAdditionalCategory.availableGlobally));

    return (
      <div className="page page-limited" id="settings-page">
        <Suggestions suggestions="settings" />
        <Helmet defer={false} title={translate('settings.page')} />

        <PageHeader component={this.props.component} />

        <div className="side-tabs-layout settings-layout">
          <div className="side-tabs-side">
            <AllCategoriesList
              component={this.props.component}
              defaultCategory={this.props.defaultCategory}
              selectedCategory={selectedCategory}
            />
          </div>
          <div className="side-tabs-main">
            {foundAdditionalCategory && shouldRenderAdditionalCategory ? (
              foundAdditionalCategory.renderComponent({
                component: this.props.component,
                selectedCategory: originalCategory
              })
            ) : (
              <CategoryDefinitionsList
                category={selectedCategory}
                component={this.props.component}
              />
            )}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state: Store) => ({
  defaultCategory: getSettingsAppDefaultCategory(state)
});

const mapDispatchToProps = { fetchSettings: fetchSettings as any };

export default connect(mapStateToProps, mapDispatchToProps)(App);
