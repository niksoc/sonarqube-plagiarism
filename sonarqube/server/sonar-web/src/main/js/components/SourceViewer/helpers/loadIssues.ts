/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { searchIssues } from '../../../api/issues';
import { getBranchLikeQuery } from '../../../helpers/branch-like';
import { parseIssueFromResponse } from '../../../helpers/issues';
import { BranchLike } from '../../../types/branch-like';

// maximum possible value
const PAGE_SIZE = 500;

function buildQuery(component: string, branchLike: BranchLike | undefined) {
  return {
    additionalFields: '_all',
    resolved: 'false',
    componentKeys: component,
    s: 'FILE_LINE',
    ...getBranchLikeQuery(branchLike)
  };
}

export function loadPage(
  query: T.RawQuery,
  page: number,
  pageSize = PAGE_SIZE
): Promise<T.Issue[]> {
  return searchIssues({
    ...query,
    p: page,
    ps: pageSize
  }).then(r =>
    r.issues.map(issue => parseIssueFromResponse(issue, r.components, r.users, r.rules))
  );
}

export function loadPageAndNext(
  query: T.RawQuery,
  toLine: number,
  page: number,
  pageSize = PAGE_SIZE
): Promise<T.Issue[]> {
  return loadPage(query, page).then(issues => {
    if (issues.length === 0) {
      return [];
    }

    const lastIssue = issues[issues.length - 1];

    if (
      (lastIssue.textRange != null && lastIssue.textRange.endLine > toLine) ||
      issues.length < pageSize
    ) {
      return issues;
    }

    return loadPageAndNext(query, toLine, page + 1, pageSize).then(nextIssues => {
      return [...issues, ...nextIssues];
    });
  });
}

export default function loadIssues(
  component: string,
  _fromLine: number,
  toLine: number,
  branchLike: BranchLike | undefined
): Promise<T.Issue[]> {
  const query = buildQuery(component, branchLike);
  return new Promise(resolve => {
    loadPageAndNext(query, toLine, 1).then(issues => {
      resolve(issues);
    });
  });
}
