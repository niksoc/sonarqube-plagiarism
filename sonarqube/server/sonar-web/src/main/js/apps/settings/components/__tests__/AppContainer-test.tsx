/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { waitAndUpdate } from 'sonar-ui-common/helpers/testUtils';
import { mockLocation, mockRouter } from '../../../../helpers/testMocks';
import {
  ALM_INTEGRATION,
  ANALYSIS_SCOPE_CATEGORY,
  LANGUAGES_CATEGORY,
  NEW_CODE_PERIOD_CATEGORY,
  PULL_REQUEST_DECORATION_BINDING_CATEGORY
} from '../AdditionalCategoryKeys';
import { App } from '../AppContainer';

it('should render default view correctly', async () => {
  const wrapper = shallowRender();

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render newCodePeriod correctly', async () => {
  const wrapper = shallowRender({
    location: mockLocation({ query: { category: NEW_CODE_PERIOD_CATEGORY } })
  });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render languages correctly', async () => {
  const wrapper = shallowRender({
    location: mockLocation({ query: { category: LANGUAGES_CATEGORY } })
  });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render analysis scope correctly', async () => {
  const wrapper = shallowRender({
    location: mockLocation({ query: { category: ANALYSIS_SCOPE_CATEGORY } })
  });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render ALM integration correctly', async () => {
  const wrapper = shallowRender({
    location: mockLocation({ query: { category: ALM_INTEGRATION } })
  });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render pull request decoration binding correctly', async () => {
  const wrapper = shallowRender({
    location: mockLocation({ query: { category: PULL_REQUEST_DECORATION_BINDING_CATEGORY } })
  });

  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

function shallowRender(props: Partial<App['props']> = {}) {
  return shallow(
    <App
      defaultCategory="general"
      fetchSettings={jest.fn().mockResolvedValue({})}
      location={mockLocation()}
      params={{}}
      router={mockRouter()}
      routes={[]}
      {...props}
    />
  );
}
