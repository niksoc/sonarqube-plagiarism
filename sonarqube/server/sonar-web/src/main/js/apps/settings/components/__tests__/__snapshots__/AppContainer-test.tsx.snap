// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ALM integration correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="almintegration"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <Connect(withAppState(AlmIntegration))
        selectedCategory="almintegration"
      />
    </div>
  </div>
</div>
`;

exports[`should render analysis scope correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="exclusions"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <AnalysisScope
        selectedCategory="exclusions"
      />
    </div>
  </div>
</div>
`;

exports[`should render default view correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="general"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <Connect(SubCategoryDefinitionsList)
        category="general"
      />
    </div>
  </div>
</div>
`;

exports[`should render languages correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="languages"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <withRouter(Connect(Languages))
        selectedCategory="languages"
      />
    </div>
  </div>
</div>
`;

exports[`should render newCodePeriod correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="new_code_period"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <NewCodePeriod />
    </div>
  </div>
</div>
`;

exports[`should render pull request decoration binding correctly 1`] = `
<div
  className="page page-limited"
  id="settings-page"
>
  <Suggestions
    suggestions="settings"
  />
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="settings.page"
  />
  <PageHeader />
  <div
    className="side-tabs-layout settings-layout"
  >
    <div
      className="side-tabs-side"
    >
      <Connect(CategoriesList)
        defaultCategory="general"
        selectedCategory="pull_request_decoration_binding"
      />
    </div>
    <div
      className="side-tabs-main"
    >
      <Connect(SubCategoryDefinitionsList)
        category="pull_request_decoration_binding"
      />
    </div>
  </div>
</div>
`;
